src/command_modules/azure-cli-interactive/azclishell/_dump_commands.py

```diff
@@ -18,122 +18,116 @@
                         'required': required,
                         'help': help_desc
                     }
-                    data[cmd]['parameters'] = {
-                        param["name"].split()[0]: options
+                    param_descrip[self.command_table[cmd].arguments[key].options_list[0]] = options
+
+                com_descrip['parameters'] = param_descrip
+                data[cmd] = com_descrip
+            except (ImportError, ValueError):
+                pass
+
+        for cmd in helps:
+            diction_help = yaml.load(helps[cmd])
+            if "short-summary" in diction_help:
+                if cmd in data:
+                    data[cmd]['help'] = diction_help["short-summary"]
+                else:
+                    data[cmd] = {
+                        'help': diction_help["short-summary"],
+                        'parameters': {}
                     }
-                if "short-summary" in param:
-                    data[cmd]['parameters'][param["name"].split()[0]]['help']\
-                        = param["short-summary"]
-        if "examples" in diction_help:
-            examples = []
-            for example in diction_help["examples"]:
-                examples.append([example['name'], example['text']])
-            data[cmd]['examples'] = examples
-
-    with open(os.path.join(get_cache_dir(), command_file), 'w') as help_file:
-        json.dump(data, help_file)
-
-
-class Exporter(json.JSONEncoder):
-    def default(self, o):
-        try:
-            return super(Exporter, self).default(o)
-        except TypeError:
-            return str(o)
+                if callable(data[cmd]['help']):
+                    data[cmd]['help'] = data[cmd]['help']()
+
+            if cmd not in data:
+                print("Command: {} not in Command Table".format(cmd))
+                continue
+
+            if "parameters" in diction_help:
+                for param in diction_help["parameters"]:
+                    if param["name"].split()[0] not in data[cmd]['parameters']:
+                        options = {
+                            'name': name_options,
+                            'required': required,
+                            'help': help_desc
+                        }
+                        data[cmd]['parameters'] = {
+                            param["name"].split()[0]: options
+                        }
+                    if "short-summary" in param:
+                        data[cmd]['parameters'][param["name"].split()[0]]['help']\
+                            = param["short-summary"]
+            if "examples" in diction_help:
+                examples = []
+                for example in diction_help["examples"]:
+                    examples.append([example['name'], example['text']])
+                data[cmd]['examples'] = examples
+
+        with open(os.path.join(get_cache_dir(), command_file), 'w') as help_file:
+            json.dump(data, help_file)
```

Comment:
Will concurrency be an issue? Especially you're doing command table initialization in a thread.